### PR TITLE
Multiview KMeans Clustering

### PR DESCRIPTION
**Pull request by achang63**
This pull request includes the multi-view clustering algorithm, which is an extension of the kmeans clustering algorithm to multiple views using the co-training framework. The PR also includes unit tests, validation notebooks, and tutorial notebooks for the algorithm.

### CODE REVIEW

File: multiview/cluster/mv_k_means.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+        Xs = check_Xs(Xs, enforce_views=2)
+
+        # Run multi-view kmeans for n_init different centroid initializations
+        min_inertia = np.inf
+        best_centroids = None
+
+        for _ in range(self.n_init):
+
+            # Random initialization of centroids
```
Comment (bdpedigo) [inline multiview/cluster/mv_k_means.py]: just so you know, this is a huge disadvantage for your alg vs. initializing with k-means++ and in my experience matters a lot. not going to make you look into it now, but I'd make an issue at least

File: multiview/cluster/mv_k_means.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+        if not (isinstance(n_init, int) and (n_init > 0)):
+            msg = 'n_init must be a nonnegative integer'
+            raise ValueError(msg)
+
+        self.n_clusters = n_clusters
+        self.random_state = random_state
+        self.patience = patience
+        self.n_init = n_init
+        self.centroids_ = None
```
Comment (bdpedigo) [inline multiview/cluster/mv_k_means.py]: don't need this I don't think

File: multiview/cluster/mv_k_means.py (hunk 1 above)
Comment (bdpedigo) [reply multiview/cluster/mv_k_means.py]: also, consider in the future adding option for custom initialization

**Comment by bdpedigo** [comment]
- I'd prefer to see a plot or two of the simulation settings that you are describing in the validation notebook.
- is there anything in the paper to benchmark against?

File: multiview/cluster/mv_k_means.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+            raise AttributeError(msg)
+
+        dist1 = cdist(Xs[0], self.centroids_[0])
+        dist2 = cdist(Xs[1], self.centroids_[1])
+        dist_metric = dist1 + dist2
+        predictions = np.argmin(dist_metric, axis=1).flatten()
+
+        return predictions
+
+    def fit_predict(self, Xs):
```
Comment (bdpedigo) [inline multiview/cluster/mv_k_means.py]: this should be in base I think

File: multiview/cluster/mv_k_means.py (hunk 1 above)
Comment (achang63) [reply multiview/cluster/mv_k_means.py]: Will do

File: multiview/cluster/mv_k_means.py (hunk 3 above)
Comment (achang63) [reply multiview/cluster/mv_k_means.py]: Do you mean the entire implementation for the fit_predict method should be in base class?

File: multiview/cluster/mv_k_means.py (hunk 2 above)
Comment (achang63) [reply multiview/cluster/mv_k_means.py]: Do you mean the centroids attribute? I think the centroids need to be saved as an attribute during training in order to use them during predictions.

**Pull request merged**